src/features/import-map.js

```diff
@@ -10,40 +10,68 @@
 import { baseUrl as pageBaseUrl, parseImportMap, resolveImportMap } from '../common.js';
 import { systemJSPrototype } from '../system-core.js';
 
-var importMap, importMapPromise;
-if (typeof document !== 'undefined') {
-  const scripts = document.getElementsByTagName('script');
-  for (let i = 0; i < scripts.length; i++) {
-    const script = scripts[i];
-    if (script.type !== 'systemjs-importmap')
-      continue;
-
-    if (!script.src) {
-      importMap = parseImportMap(JSON.parse(script.innerHTML), pageBaseUrl);
-    }
-    else {
-      importMapPromise = fetch(script.src)
-      .then(function (res) {
-        return res.json();
-      })
-      .then(function (json) {
-        importMap = parseImportMap(json, script.src);
-      });
+var acquiringImportMaps = true, importMapPromise;
+
+function retrieveImportMaps() {
+  if (typeof document !== 'undefined') {
+    const scripts = document.getElementsByTagName('script');
+    let pendingMaps = []
+
+    for (let i = 0; i < scripts.length; i++) {
+      const script = scripts[i];
+      if (script.type !== 'systemjs-importmap')
+        continue;
+
+      if (!script.src) {
+        pendingMaps.push(parseImportMap(JSON.parse(script.innerHTML), pageBaseUrl));
+      }
+      else {
+        pendingMaps.push(
+          fetch(script.src)
+          .then(function (res) {
+            return res.json();
+          })
+          .then(function (data) {
+            return parseImportMap(data, script.src);
+          })
+        );
+      }
     }
-    break;
+
+    const initialImportMap = { imports: {}, scopes: {} };
+
+    importMapPromise = Promise.all(pendingMaps).then(function(importMaps) {
+      return importMaps.reduce(function(finalMap, importMap) {
+        return mergeImportMap(finalMap, importMap);
+      }, initialImportMap);
+    });
   }
 }
 
-importMap = importMap || { imports: {}, scopes: {} };
+function mergeImportMap(originalMap, newMap) {
+  for (let i in newMap.imports) {
```

Comment:
The import maps specification indicates that merging is roughly equivalent to `{...firstMap, ...secondMap}`. Since the rest operator isn't supported in IE11 (nor is Object.assign), I implemented this with for loops. I'm also open to using the rest operator and letting babel compile it down with a helper function.

Reply:
Yup, compatibility is the goal so a loop is fine.